src/System.Drawing.Common/src/System/Drawing/SystemIcons.cs

```diff
@@ -212,5 +72,15 @@ public static Icon Shield
                 return s_shield;
             }
         }
+
+        private static Icon GetIcon(Icon icon, int iconId)
+        {
+            if (icon == null)
+            {
+                icon = new Icon(SafeNativeMethods.LoadIcon(NativeMethods.NullHandleRef, SafeNativeMethods.IDI_WINLOGO));
```

Comment:
This is broken; nothing here is getting cached.  You need to change it to pass the icon with ref, e.g.
```C#
private static Icon GetIcon(ref Icon icon, int iconId) =>
    icon ?? (icon = new Icon(...));
```